src/pages/en/guides/migrate-to-astro/from-gatsby.mdx

```diff
@@ -2,67 +2,884 @@
+
+:::tip
+Prefer to explore in your browser instead? Visit https://astro.new for a variety of starter templates that can be opened and edited in StackBlitz, CodeSandbox or Gitpod.
+:::
+
+### Install integrations (optional)
+
+You may find it useful to install some of [Astro's optional integrations](/en/guides/integrations-guide/) to use while converting your Gatsby project to Astro:
+
+- **@astrojs/react**: to reuse some existing React UI components in your new Astro site, or keep writing with React components.
+
+- **@astrojs/image**: to replace Gatsby's Image plugin with Astro's own image-optimizing components.
+
+- **@astrojs/mdx**: to bring existing MDX files from your Gatsby project, or to use MDX in your new Astro site.
+
+### Put your code in `src`
+
+1. **Delete** Gatsby's `public/` folder. 
+    
+    Gatsby uses the `public/` directory for its build output, so you can safely discard this folder. You will no longer need a built version of your Gatsby site. (Astro uses `dist/` by default for the build output.)
+
+2. **Rename** Gatsby's `static/` folder to `public/`, and use it as Astro's `public/` folder. 
+
+    Astro uses a folder called `public/` for static assets. You can alternatively copy the contents of `static/` into an existing Astro `public/` folder.
+
+3. **Copy or Move** Gatsby's other files and folders (e.g. `components`, `pages` etc.) into Astro's `src/` folder.
+
+  Astro's `src/pages/` folder is a special folder used for file-based routing to create your site's pages and posts from `.astro`, `.md` and `.mdx` files. You will not have to configure any routing behavior for your Astro, Markdown and MDX files.
+
+  All other folders are optional, and you can organize the contents of your `src/` folder any way you like. Other common folders in Astro projects include `src/layouts/`, `src/components`, `src/styles`, `src/scripts`.
+
+### Repurpose config files
+
+Gatsby has several top-level configuration files for configuration options, site and page metadata and generating page routes. You will not use any of these `gatsby-*.js` files in your Astro project, but there may be some content that you can reuse:
+
+- `gatsby-config.js`: Move your `siteMetadata: {}` into `src/data/siteMetadata.js` (or `siteMetadata.json`) to import data about your site (title, description, social accounts etc.) into page layouts.
+
+- `gatsby-browser.js`: Consider adding anything used here directly into your main layout's `<head>` tag.
+
+- `gatsby-node.js`: You will not need to create your own nodes in Astro, but viewing the schema in this file may help you with defining types in your Astro project.
+
+- `gatsby-ssr.js`: If you choose to use SSR in Astro, you will add and configure the adapter of your choice directly in `astro.config.mjs`.
+
+### Convert JSX files to `.astro` files
+
+Here are some tips for converting a Gatsby `.js` component into a `.astro` component:
```

Comment:
I think this section can be more clear in a recommendation on how to use React vs. Astro.

And it should also have a solution on how to handle images (see my comment above).

My take is:

- You can keep your JSX and just update imports and Gatsby specific components like `<Link>`. Only the index.astro needs to be an .astro file. (That is correct, right?)
  - Again, in this case: How to handle images?
- However, you may want to embrace astro a bit more. For example use .astro files for alle content that is static and JSX files only whenever content needs to be interactive for the client (islands). This will remove a lot of JSX boilderplate and give you full access to great astro features like glob (imports?).

Its important to understnad the barriers between React and Astro:
- You can import a react ocmponent in a .astro file.  (Link)
- You can even mix astro components and react components as long as this is in a .astro file (Link)
- However, you cannot import a .astro component in a .jsx file. Which means some helper components might need to be duplicates or build using react to make the compatible (see my comment in the other ticket…)

Reply:
Update: I see in https://github.com/withastro/docs/pull/2211/files#diff-1c76fd47c1d1b25a4ed7614df83f293191bed75840838ebe5fd4ac61d157869cR262, that the part about pages above is wrong, so al pages need to become .astro files…